analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantInheritanceList.cs

```diff
@@ -229,7 +206,11 @@ private static Location GetLocationWithToken(TypeSyntax type, SeparatedSyntaxLis
             return Location.Create(type.SyntaxTree, new TextSpan(start, end - start));
         }
 
-        private static bool IsBaseListNullOrEmpty(BaseTypeDeclarationSyntax baseTypeDeclaration) =>
-            baseTypeDeclaration.BaseList == null || !baseTypeDeclaration.BaseList.Types.Any();
+        private static ImmutableDictionary<string, string> DiagnosticsProperties(int redundantIndex)
```

Comment:
The original trick with `.Empty.Add(..)` could do this in a single line and arrow function.

Reply:
That's right. I used the builder intentionally because it just scales better. If we ever need to store more than one property, `.Add` is just inefficient and the builder is the preferred way to create the dictionary anyway. See also the thread at https://stackoverflow.com/a/42094674

Reply:
Once we reach the need, we can rewrite it. It's not needed for what we need now, and I don't expect changes any time soon.